### PR TITLE
SPRK-2842 SPRK-2969 commit existing index to code

### PR DESCRIPTION
**Pull request by crbunney**
Enhanced the existing Dockerfile and switched to poetry as part of setting up an environment I could use to create migrations for the index change

Tested by updating Spark and getting passing builds in https://github.com/sponge-learning/spark/pull/1557

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+[tool.poetry]
+name = "lrs"
+version = "0.1.0"
+authors = []
+description = ""
+readme = "readme.md"
+packages = [{include = "lrs"}]
+
+[tool.poetry.dependencies]
+python = "~3.7.0"
```
Comment (crbunney) [inline pyproject.toml:14]: This is what was required to run the make-migrations command. I've checked this against Spark's requirements.txt to keep them consistent.

**Comment by crbunney** [comment]
Tested by updating Spark and getting passing builds in https://github.com/sponge-learning/spark/pull/1557

File: lrs/models.py [hunk 2]
```diff
@@ -312,6 +312,15 @@ def get_queryset(self):
 
 
 class Activity(models.Model):
+
+    class Meta:
+        indexes = [
+            models.Index(
+                fields=["activity_id", "canonical_version"],
+                condition=models.Q(canonical_version__isnull=False),
+                name="ix_activity_canonical_version"
```
Comment (vvm46Vasylyk) [inline lrs/models.py]: maybe here should be: **is**_activity_canonical_version  instead of `ix_` ?

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    command: python manage.py runserver 0.0.0.0:8000
+    volumes:
+      - .:/app
+      - ./lrs/static:/static
+    environment:
+      PYTHONUNBUFFERED: "1"
+
+
+
+
```
Comment (vvm46Vasylyk) [inline docker-compose.yml]: can we remove unnecessary blank lines?

**Review by vvm46Vasylyk — CHANGES_REQUESTED** [review]

File: lrs/models.py (hunk 2 above)
Comment (crbunney) [reply lrs/models.py]: It was deliberate, based on some stuff I googled about index naming conventions - `ix` for **i**nde**x**.

Thinking about it more, I should probably keep the name consistent with the manually applied index: `idx_canonical_activity` so I'll change it to that

Comment (crbunney) [reply lrs/models.py]: done

**Review by dbramwell — APPROVED** [review]

**Pull request merged**